airbyte-integrations/connectors/destination-redshift/src/test/java/io/airbyte/integrations/destination/redshift/RedshiftCopierTest.java

```diff
@@ -0,0 +1,172 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.redshift;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airbyte.db.Databases;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.db.jdbc.JdbcUtils;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.SyncMode;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@Disabled("Disabled on automatic execution as the tests requires specific credentials to run."
+    + "Intended to be a lighter-weight complement to the Docker integration tests. Fill in the appropriate blank variables to run.")
+public class RedshiftCopierTest {
+
+  // Fill in the empty variables before running.
+  private static final String S3_KEY_ID = "";
+  private static final String S3_KEY = "";
+  private static final String REDSHIFT_CONNECTION_STRING = "";
+  private static final String REDSHIFT_USER = "";
+  private static final String REDSHIFT_PASS = "";
+
+  private static final String S3_REGION = "us-west-2";
+  private static final String TEST_BUCKET = "redshift-copier-test";
+  private static final String RUN_FOLDER = "test-folder";
+  private static final String SCHEMA_NAME = "public";
+  private static final String STREAM_NAME = "redshift_copier_test";
+  private static final String RAW_TABLE_NAME = new RedshiftSQLNameTransformer().getRawTableName(STREAM_NAME);
```

Comment:
even though `getRawTableName` is deprecated - it's less confusing to use the deprecated method for now, and refactor everything at one go later. I'll put up a PR once everything is done.